pkg/metrics/registry.go

```diff
@@ -14,15 +14,44 @@
 
 package metrics
 
-/*
-import "github.com/golang/protobuf/proto"
+import (
+	"fmt"
+	"reflect"
 
-var registeredMetrics = make(map[string]func() proto.Message)
+	"github.com/ligato/vpp-agent/pkg/models"
+)
 
-func Register(pb proto.Message, fn func() proto.Message) {
-    models.
-    if _, ok := registeredMetrics; ok {
-        panic("duplicate registration for metrics %s", )
-    }
+var registeredMetrics = make(map[string]Retriever)
```

Comment:
This would need a mutex in case `Register` and `Retrieve` are called concurrently